Sources/Db/APIs/PostgreSQL.php

```diff
@@ -2258,11 +2254,15 @@ function ($errno, $errstr) {
 	 * the database.
 	 *
 	 * @param array $matches The matches from preg_replace_callback
+	 * @param array $db_values = array() The values to be inserted into the string
+	 * @param null|object $connection = null The connection to use (null to use $db_connection)
 	 * @return string The appropriate string depending on $matches[1]
 	 */
-	protected function replacement__callback(array $matches): string
+	protected function replacement__callback(array $matches, array $db_values, ?object $connection = null): string
 	{
-		if (!is_object($this->temp_connection)) {
+		$connection ??= $this->connection;
+
+		if (!is_object($connection)) {
```

Comment:
```suggestion
	 * @param array $db_values The values to be inserted into the string.
	 * @param object $connection The connection to use.
	 * @return string The appropriate string depending on $matches[1].
	 */
	protected function replacement__callback(array $matches, array $db_values, object $connection): string
{
```

Since this method is only called internally, we know that we will always pass a non-null $connection value.

Reply:
Oops. Don't just merge this suggestion. I meant to include the following three lines in the selection for the suggestion. They should be removed along with the lines that are removed in this suggestion.